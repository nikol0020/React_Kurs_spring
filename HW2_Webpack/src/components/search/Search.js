import React, {Component} from "react";
import './Search.scss';
import Button from '../button';
import '../propTypes';

export default class Search extends Component {

    constructor(props) {
        super(props);
        this.state = {
            filter: 'TITLE',
            searchTxt: ''
        };
    };


    changeFilter = e => {
        this.setState({filter: e.target.value});
    };

    changeSearchTxt = e => {
        const filteredData = this.props.setFilteredData(this.state.filter);
        this.setState({searchTxt: e.target.value});
        this.props.setFilteredData ( filteredData);
    };

    render() {
        return (
            <div className="search__box">
                <span className="search__text">FIND YOUR MOVIE</span>
                <br/>
                <input className="search__input" id="searchInput" type="text" name="searchInput" placeholder="SEARCH" onChange={this.changeSearchTxt}/>
                <hr/>
                <div className="search__buttons">
                    <div className="search__buttons-wrapper">
                        <span>SEARCH BY</span>
                        <Button
                            value="TITLE"
                            active={"TITLE" === this.state.filter}
                            onClick={this.changeFilter}
                            id="buttonTitle"
                        />
                        <Button
                            value="GENRE"
                            active={"GENRE" === this.state.filter}
                            onClick={this.changeFilter}
                            id="buttonGenre"
                        />
                    </div>
                    <Button
                        value="SEARCH"
                        active={"SEARCH" === this.state.filter}
                        onClick={this.changeFilter}
                        />
                </div>
                <div className="search__count">
                    <span className="search__count-qty"></span>
                    <span className="search__text">movies found</span>
                </div>
            </div>
        )
    }
};

