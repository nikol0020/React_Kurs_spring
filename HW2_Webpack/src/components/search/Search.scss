.search {

  &__input {
    width: 100%;
    height: 30px;
    margin: 10px auto;
    border-radius: 5px;
  }

  &__buttons {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
  }

  &__box {
    width: 80%;
    margin: 0 10%;
  }

  &__count {
    width: 300px;
    min-width: 300px;
    min-height: 50px;
    margin-top: 25px;
    float: left;
  }

  &__buttons-wraper {
    font-size: 12px;

    & > input {
      margin: 0 10px;
    }
  }

  &__button {
    width: 70px;
    height: 30px;
    border-radius: 5px;

    &:hover {
      cursor: pointer;
    }
  }

  &__count-qty {
    margin-right: 15px;
  }

}
