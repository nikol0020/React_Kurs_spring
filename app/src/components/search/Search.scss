.search {

  &__input {
    width: 100%;
    height: 30px;
    margin: 10px auto;
    border-radius: 5px;
    font-size: 16px;
  }

  &__buttons {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
  }

  &__box {
    width: 80%;
    margin: 0 10%;
  }

  &__count {
    width: 300px;
    min-width: 300px;
    min-height: 50px;
    margin-top: 40px;
    float: left;
    position: relative
  ;
    z-index: 1;
    color: #888;
  }

  &__buttons-wrapper {
    font-size: 12px;

    & > input {
      margin: 0 10px;
    }
  }

  &__count-qty {
    margin-right: 15px;
  }

}
