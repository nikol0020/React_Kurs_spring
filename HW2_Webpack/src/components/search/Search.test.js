import React from 'react';
import Search from './Search';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('<Search /> shallow rendering', () => {

    it('should render a div', () => {
        const wrapper = shallow(
            <div className="search__box"></div>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('<span>FIND YOUR MOVIE</span>', () => {
        const wrapper = shallow(
            <span className="search__text">FIND YOUR MOVIE</span>
        );
        expect(wrapper.find('span').text()).toBe('FIND YOUR MOVIE');
    });

});


describe('when click buttonGenre', () => {
    const valueFilter = 'TITLE';
    const wrapper = shallow(<Search />);

    beforeEach(() => {
        wrapper.find('#buttonTitle').simulate('click', {
            target: {
                value: valueFilter
            },
        });
    });

    it('updates filter field in state', () => {
        expect(wrapper.state().filter).toEqual(valueFilter)
    })
});

describe('when click buttonGenre', () => {
    const valueFilter = 'GENRE';
    const wrapper = shallow(<Search />);

    beforeEach(() => {
        wrapper.find('#buttonGenre').simulate('click', {
            target: {
                value: valueFilter
            },
        });
    });

    it('updates filter field in state', () => {
        expect(wrapper.state().filter).toEqual(valueFilter)
    })
});
