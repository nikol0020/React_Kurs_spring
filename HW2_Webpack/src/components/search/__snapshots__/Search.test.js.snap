// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> shallow rendering should render a div 1`] = `
<div
  className="search__box"
/>
`;
